app/src/util/navigation.ts

```diff
@@ -6,15 +6,20 @@ export const navigationRef = createNavigationContainerRef<StackParamList>();
 
 export async function getNavigate() {
   await new Promise<void>((resolve) => {
-    const poll = () => {
+    const poll = (): boolean => {
```

Comment:
what is this `poll` for?

Reply:
This poll is to wait for the navigationRef to be populated correctly, as there is no event or Promise from the library that we can depend on. If we try to use the navigationRef without polling, we'll get race conditions.